src/lib/storage/dictionary_column.hpp

```diff
@@ -21,51 +25,115 @@ constexpr ValueID INVALID_VALUE_ID{std::numeric_limits<ValueID::base_type>::max(
 template <typename T>
 class DictionaryColumn : public BaseColumn {
  public:
+
+  DictionaryColumn() = delete;
+
   /**
    * Creates a Dictionary column from a given value column.
    */
-  explicit DictionaryColumn(const std::shared_ptr<BaseColumn>& base_column);
+  explicit DictionaryColumn(const std::shared_ptr<BaseColumn>& base_column) : _dictionary(nullptr), _attribute_vector(nullptr) {
+    std::set<T> sorter;
+    for(size_t i = 0; i < base_column->size(); ++i) {
+      sorter.insert(type_cast<T>((*base_column)[i]));
+    }
+
+    _dictionary = std::make_shared<std::vector<T>>();
+    _dictionary->reserve(sorter.size());
+    std::copy(sorter.begin(), sorter.end(), std::back_inserter(*_dictionary));
+
+    if(sorter.size() > std::numeric_limits<uint32_t>::max()) {
+      _attribute_vector = std::make_shared<FittedAttributeVector<uint64_t>>();
+    }
+    else if (sorter.size() > std::numeric_limits<uint16_t>::max()) {
+      _attribute_vector = std::make_shared<FittedAttributeVector<uint32_t>>();
+    }
+    else if (sorter.size() > std::numeric_limits<uint8_t>::max()) {
+      _attribute_vector = std::make_shared<FittedAttributeVector<uint16_t>>();
+    }
+    else {
+      _attribute_vector = std::make_shared<FittedAttributeVector<uint8_t>>();
+    }
```

Comment:
line 44-55
I prefer to encapsulate this into an extra function for better readability.
=>
_attribute_vector = _create_attribute_vector(sorter.size());

Reply:
sounds good.